src/Reactor.cc

```diff
@@ -479,6 +479,41 @@ void Reactor::switchMachine(const string& machine)
 	deleteBoard(oldBoard);
 }
 
+void Reactor::switchMachineFromSetup(const string& filename)
+{
+	if (!display) {
+		display = make_unique<Display>(*this);
+		// TODO: Currently it is not possible to move this call into the
+		//       constructor of Display because the call to createVideoSystem()
+		//       indirectly calls Reactor.getDisplay().
+		display->createVideoSystem();
+	}
+
+	// create new machine
+	// load state into machine
+	// switch to new machine
+	// delete old active machine
+
+	assert(Thread::isMainThread());
+	auto newBoard = createEmptyMotherBoard();
+
+	try {
+		XmlInputArchive in(filename);
+		in.serialize("machine", *newBoard);
+	} catch (XMLException& e) {
+		throw CommandException("Cannot load setup, bad file format: ",
+				       e.getMessage());
+	} catch (MSXException& e) {
+		throw CommandException("Cannot load setup: ", e.getMessage());
+	}
+
+	boards.push_back(newBoard);
+
+	auto oldBoard = activeBoard;
+	switchBoard(newBoard);
+	deleteBoard(oldBoard);
+}
```

Comment:
I'm not saying this is wrong. Or that it should be done in a different way. But this code surprises me.

In my mind setup files are just savestates (at least internally at a technical level, for the user it are different concepts). And we do already have code to load savestates. So I'm surprised to see new code to load a setup. (I did not actually compare this code with the code that loads a savestate .. maybe there are substantial differences??).

Reply:
We have no C++ code to load savestates, we do that mostly from Tcl with more low level commands. But we also don't have a command line option to load a savestate. And for the handling of the `-setup` command line option, I do need that C++ code. There is some of this code in the implementation of some commands (like RestoreMachineCommand), but it seemed to small to start reusing. Therefore I just wrote a new method for this specific purpose, similar to the `-machine` option (also regarding the `machine` vs `setup` command).

Reply:
@m9710797 are you satisfied enough about this code to close the thread?